internal/nix/shell.go

```diff
@@ -237,7 +238,7 @@ func (s *Shell) Run(nixShellFilePath string) error {
 	cmd.Stderr = os.Stderr
 
 	debug.Log("Executing nix-shell command: %v", cmd.Args)
-	return errors.WithStack(cmd.Run())
+	return errors.WithStack(usererr.NewExecError(cmd.Run()))
```

Comment:
This is a bit messy, but I believe this can be invoked for two scenarios:
1. regular `devbox shell`
2. running a devbox script.

I think we want `usererro.NewExecError` only for the scenario (2).

To fix that we could wrap this here by checking `if s.ScriptCommand != "" {`
(and for the if-flakes scenario above too)

Reply:
Will do. Ugh this logic is getting quite convoluted. cc @ipince please launch your print-dev-env stuff soon 😢